File: regional/regional/2015/I_O/Excellence/Excellence.java
import java.io.*;
import java.util.*;
public class Excellence {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		PrintWriter pw = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
		int n = Integer.parseInt(br.readLine());
		int[] list = new int[n];
		for(int i = 0; i < n; i++) {
			list[i] = Integer.parseInt(br.readLine());
		}
		Arrays.sort(list);
		int ret = Integer.MAX_VALUE;
		for(int i = 0; i < n-1-i; i++) {
			ret = Math.min(ret, list[i] + list[n-1-i]);
		}
		pw.println(ret);
		pw.close();
	}
}

File: regional/PacificNW/2015/I_O/Excellence/Excellence.pl
scalar <> ;
while (<>) {
   chomp ;
   push @a, $_ ;
}
@a = sort { $a <=> $b } @a ;
$min = $a[0] + $a[@a-1] ;
for ($i=0; $i<@a; $i++) {
   $t = $a[$i] + $a[@a-$i-1] ;
   $min = $t if $t < $min ;
}
print "$min\n" ;
